app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt

```diff
@@ -148,9 +148,16 @@ class BrowserFragment : IWebViewLifecycleFragment() {
                 view = layout.cursorView)
         lifecycle.addObserver(cursor!!)
 
-        layout.browserOverlay.onNavigationEvent = onNavigationEvent
-        layout.browserOverlay.navigationStateProvider = NavigationStateProvider()
-        layout.browserOverlay.visibility = overlayVisibleCached ?: View.GONE
+        with (layout.browserOverlay) {
+            onNavigationEvent = this@BrowserFragment.onNavigationEvent
+            navigationStateProvider = NavigationStateProvider()
+            visibility = overlayVisibleCached ?: View.GONE
+            onPreSetVisibilityListener = { isVisible ->
```

Comment:
I had forgotten that this was called in `setVisible`. Maybe add a comment in the declaration of this variable?